### PR TITLE
Tabular: Raise Exception if duplicate indices present in data

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*
- Raise exception if duplicate indices in data.
- Fixes incorrect prediction dimensions returned in `predictor.get_oof_pred_proba` when training data contained duplicate indices.
- Fixed incorrect eval metric comparison in pseudo labeling logic (only impacts warning message).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1395-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1395/1/index.html

**Review by yinweisu — APPROVED** [review]

**Review by gradientsky — APPROVED** [review]
